packages/client/src/client.ts

```diff
@@ -14,6 +36,16 @@ export type CreateClientOptions<TContract extends ContractDefinition> = {
   urls: ConnectionUrl[];
   connectionOptions?: AmqpConnectionManagerOptions | undefined;
   logger?: Logger | undefined;
+  /**
+   * Optional OpenTelemetry instrumentation for tracing
+   * Requires @amqp-contract/opentelemetry package
+   */
+  instrumentation?: ClientInstrumentation | undefined;
+  /**
+   * Optional OpenTelemetry metrics for monitoring
+   * Requires @amqp-contract/opentelemetry package
+   */
+  metrics?: ClientMetrics | undefined;
```

Comment:
would be preferable to require instrumentation config, same for metrics and let the client instanciate them (instead of give the responsability to the end user)